src/coreclr/nativeaot/Runtime/thread.cpp

```diff
@@ -1323,6 +1340,43 @@ FCIMPL0(size_t, RhGetDefaultStackSize)
 }
 FCIMPLEND
 
+#ifdef TARGET_WINDOWS
+// Native APC callback for Thread.Interrupt
+// This callback sets the interrupt flag on the current thread
+static VOID CALLBACK InterruptApcCallback(ULONG_PTR /* parameter */)
+{
+    // Set the interrupt flag on the current thread
+    // If we were queued on a thread that wasn't started,
+    // we may be the first code that runs on this thread.
+    // Ensure we're attached to the thread store.
+    ThreadStore::AttachCurrentThread();
```

Comment:
Do we need to check for detached thread and skip attaching in that case? `hreadStore::AttachCurrentThread` is going to fail fast on detached threads.